packages/core/src/factory.ts

```diff
@@ -113,19 +113,24 @@ export function factory<Resource, Input, Output, Config, LoadOpts>(
 
               return TaskEither.left(new Error('Nor forwarder nor mocker has been selected. Something is wrong'));
             }),
-            TaskEither.map(({ output, resource }) => {
-              let outputValidations: IPrismDiagnostic[] = [];
-              if (resource && components.validator && components.validator.validateOutput) {
-                outputValidations = components.validator.validateOutput(
-                  {
-                    resource,
-                    output,
-                    config: configObj,
-                  },
-                  defaultComponents.validator,
-                );
-              }
+            TaskEither.chain(({ output, resource }) => {
+              const outputValidations =
+                resource && components.validator && components.validator.validateOutput
+                  ? components.validator.validateOutput(
+                      {
+                        resource,
+                        output,
+                        config: configObj,
+                      },
+                      defaultComponents.validator,
+                    )
+                  : [];
 
+              return outputValidations.length
+                ? TaskEither.left(outputValidations[0] as any)
```

Comment:
Nice stuff! You're getting into it!

1. Why are you returning only the first element?
2. We should really avoid the `as any` and convert the output validation in a real error somehow, otherwise this is probably not going to work.

Reply:
@XVincentX, I think that when input is validated, it also returns one (first?) element. So that, at the end, there is `throw e` in that last `Left` handling function. So I thought this must be intentional (?).